Batteries/Data/ByteSlice.lean

```diff
@@ -0,0 +1,91 @@
+
+/-
+Copyright (c) 2021 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+
+set_option autoImplicit true
+
+namespace Nat
+
+/- Up -/
+
+/-- A well-ordered relation for "upwards" induction on the natural numbers up to some bound `ub`. -/
+def Up (ub a i : Nat) := i < a ∧ i < ub
+
+theorem Up.next {ub i} (h : i < ub) : Up ub (i+1) i := ⟨Nat.lt_succ_self _, h⟩
+
+theorem Up.WF (ub) : WellFounded (Up ub) :=
+  Subrelation.wf (h₂ := (measure (ub - ·)).wf) fun ⟨ia, iu⟩ => Nat.sub_lt_sub_left iu ia
+
+/-- A well-ordered relation for "upwards" induction on the natural numbers up to some bound `ub`. -/
+def upRel (ub : Nat) : WellFoundedRelation Nat := ⟨Up ub, Up.WF ub⟩
+
+end Nat
```

Comment:
This is out of place and it should be a natural byproduct of `Nat.find` but now I see that `Nat.find` is missing! Is it now called something else?

Reply:
`Nat.find` is in Mathlib, so would need to be upstreamed first

Reply:
I think this definition is not needed any more, it was originally added as a workaround for `termination_by` before it got good enough to just use subtraction as a decreasing metric.

Reply:
Concurred. Please delete since it is no longer needed.